sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java

```diff
@@ -319,38 +319,59 @@ protected String getLoggerTag() {
             final Class<T> documentType,
             final ReadOptions readOptions) {
         final DefaultAppCenterFuture<Document<T>> result = new DefaultAppCenterFuture<>();
-        if (mNetworkStateHelper.isNetworkConnected()) {
-            getTokenAndCallCosmosDbApi(
-                    partition,
-                    result,
-                    new TokenExchangeServiceCallback() {
-
-                        @Override
-                        public void callCosmosDb(TokenResult tokenResult) {
-                            callCosmosDbReadApi(tokenResult, documentId, documentType, result);
-                        }
+        postAsyncGetter(new Runnable() {
 
-                        @Override
-                        public void completeFuture(Exception e) {
-                            Storage.this.completeFuture(e, result);
+            @Override
+            public void run() {
+                Document<T> cachedDocument;
+                boolean fetchRemote = false;
+                String storedPartitionName = appendAccountIdToPartitionName(partition);
```

Comment:
This method should be addressed more friendly...

Reply:
please do not merge!

Reply:
nvm.. seems gonna a be a large refactory and a rare corner case, will be better to address in another pr.